test/ComponentInstallerTest.php

```diff
@@ -43,6 +72,426 @@ public function createApplicationConfig($contents = null)
+            }
+
+            if (! strstr($argument[2], 'application.config.php')) {
+                return false;
+            }
+
+            return true;
+        }), 0)->willReturn(1);
+
+        $this->io->ask(Argument::that(function ($argument) {
+            if (! is_array($argument)) {
+                return false;
+            }
+            if (! strstr($argument[0], 'Remember')) {
+                return false;
+            }
+
+            return true;
+        }), 'n')->willReturn('n');
+
+        $this->io->write(Argument::that(function ($argument) {
+            return strstr($argument, 'Installing SomeComponent from package some/component');
+        }))->shouldBeCalled();
+
+        $this->io->write(Argument::that(function ($argument) {
+            return strstr($argument, 'Dependency SomeDependency is not registered in the configuration');
+        }))->shouldBeCalled();
+
+        $this->assertNull($this->installer->onPostPackageInstall($event->reveal()));
+    }
+
+    public function dependency()
+    {
+        return [
+            // 'description' => [
+            //   'package name to install',
+            //   [enabled modules],
+            //   [dependencies],
+            //   [result: enabled modules in order],
+            //   autoloading: psr-0 or psr-4
+            // ],
+            'one-dependency-on-top-psr-0' => [
+                'MyPackage1',
+                ['D1', 'App'],
+                ['D1'],
+                ['D1', 'MyPackage1', 'App'],
+                'psr-0',
+            ],
+            'one-dependency-on-bottom-psr-0' => [
+                'MyPackage2',
+                ['App', 'D1'],
+                ['D1'],
+                ['App', 'D1', 'MyPackage2'],
+                'psr-0',
+            ],
+            'no-dependencies-psr-0' => [
+                'MyPackage3',
+                ['App'],
+                [],
+                ['MyPackage3', 'App'],
+                'psr-0',
+            ],
+            'two-dependencies-psr-0' => [
+                'MyPackage4',
+                ['D1', 'D2', 'App'],
+                ['D1', 'D2'],
+                ['D1', 'D2', 'MyPackage4', 'App'],
+                'psr-0',
+            ],
+            'two-dependencies-in-reverse-order-psr-0' => [
+                'MyPackage5',
+                ['D2', 'D1', 'App'],
+                ['D1', 'D2'],
+                ['D2', 'D1', 'MyPackage5', 'App'],
+                'psr-0',
+            ],
+            'two-dependencies-with-more-packages-psr-0' => [
+                'MyPackage6',
+                ['D1', 'App1', 'D2', 'App2'],
+                ['D1', 'D2'],
+                ['D1', 'App1', 'D2', 'MyPackage6', 'App2'],
+                'psr-0',
+            ],
+            // PSR-4 autoloading
+            'one-dependency-on-top-psr-4' => [
+                'MyPackage11',
+                ['D1', 'App'],
+                ['D1'],
+                ['D1', 'MyPackage11', 'App'],
+                'psr-4',
```

Comment:
This is fine. We want to ensure that the behavior is the same for both PSR-0 and PSR-4 style packages, which means we'll have tests that are essentially the same, just with the different autoloading rules.

Per an earlier comment of mine, also add test cases for usage of `files` autoloading of the `Module.php` file, however.